### PR TITLE
Add CodeQL security scan for KM (only) to CI workflow

### LINKED ISSUE #1510: Enable CodeQL scanning on KM
CodeQL is a static analysis scanner newly available from github. It is free for Open Source projects. We should use it.

### PR DESCRIPTION
**Pull request by johnmuth81**
Update workflow to include CodeQL extended security scan as part of regular CI flow.

Fixes #1510

### CODE REVIEW

**Comment by paulpopelka** [comment]
In the initializing codeql step there is a warning:

Warning: 1 issue was detected with this workflow: Using on.push.paths-ignore can prevent Code Scanning annotating new alerts in your pull requests.

Is this expected?

File: .github/workflows/km-ci-workflow.yaml [hunk 1]
```diff
@@ -514,6 +514,41 @@ jobs:
...
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+        # TODO: Enable enhanced security checks.
+        # queries: security-extended
+
+    - run: |
+       make -C km all
```
Comment (gnode1) [inline .github/workflows/km-ci-workflow.yaml:547]: Maybe -j to speed things up?

**Review by gnode1 — APPROVED** [review]
a question and a suggestion

File: .github/workflows/km-ci-workflow.yaml [hunk 2]
```diff
@@ -514,6 +514,41 @@ jobs:
...
+      contents: read
+      security-events: write
+
+    container:
+      image: kontainapp/buildenv-km-fedora:latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp' ]
```
[the remark this answers is not in the document]
Comment (johnmuth81) [reply .github/workflows/km-ci-workflow.yaml:532]: yes, cpp as in c++. No, there isn't a separate C. C++ and C are one thing to CodeQL.

File: .github/workflows/km-ci-workflow.yaml (hunk 1 above)
Comment (johnmuth81) [reply .github/workflows/km-ci-workflow.yaml:547]: It's in parallel with everything else so it tends to be the first thing that completes, even without -j.

**Pull request merged**